src/views/projects/SourceBrowser/FileDiff.svelte

```diff
@@ -193,19 +315,56 @@
       {#if file.diff.type === "plain"}
         {#if file.diff.hunks.length > 0}
           <table class="diff">
-            {#each file.diff.hunks as hunk}
+            {#each file.diff.hunks as hunk, hunkIdx}
               <tr class="diff-line">
                 <td colspan={2} />
                 <td colspan={6} class="diff-expand-header">
                   {hunk.header}
                 </td>
               </tr>
-              {#each hunk.lines as line}
-                <tr class="diff-line" data-expanded data-type={lineSign(line)}>
-                  <td class="diff-line-number left" data-type={lineSign(line)}>
+              {#each hunk.lines as line, lineIdx}
+                {#if shouldScroll(selectedHunkLine, hunkIdx, lineIdx)}
+                  {#if !initialScroll}
+                    {(setTimeout(() => {
+                      initialScroll = true;
+                      const target = document.getElementById(
+                        file.path + "::H" + hunkIdx + "L" + lineIdx,
+                      );
+                      if (target) {
+                        target.scrollIntoView();
+                      }
+                    }, 200),
+                    "")}
+                  {/if}
+                {:else}
+                  {((initialScroll = true), "")}
```

Comment:
I don’t see a reason why this code cannot be moved to the `<script>` part of the component. Embedding imperative code in the declarative template makes it very hard to understand what is going on, especially in this case where the codes sits inside four `if` statements and a `each` iteration.